MIP/mip-16/README.md

```diff
@@ -0,0 +1,193 @@
+
+1. **execution** fees: the costs that correspond to resources usage, CPU and permanent storage.
+2. **data availability** fees: the costs of publishing transactions/blocks data to a _data availability layer_.
+3. **settlement** fees: the costs of _validating_ a transaction (e.g. zk-proof generation and verification, validators attestations.)
+
+Overall the transaction fees are defined by the sum of three types of fees:
+
+$$ {ExecFees} + {DAFees} + {SetlFees}\mathpunct. $$
+
+The fees should be expressed in a single (crypto)-currency, e.g. USD, or a token `$APT`, `$MOVE`, `$ETH`.
+
+### Execution 
+
+#### MoveVM execution fees 
+
+To execute a transaction and store the result, the operator of the chain uses some hardware: CPU and disk storge. The **execution** fees represent the compensation of the operator for executing a transaction. 
+
+Movement Network uses Aptos-Move as the execution layer and the execution fees for Aptos-Move are defined in the Aptos documentation [Gas and storage fees](https://aptos.dev/en/network/blockchain/gas-txn-fee). 
+The Aptos-Move fee mechanism splits the execution fees in two parts:
+
+- CPU and IO operations, expressed in gas units; 
+- permanent storage, expressed in `$APT`.
+
+CPU and IO operations for a transaction $tx$ result in a number of gas units $g(tx)$, and the corresponding fees are computing using a _gas price_ $GasPrice$ expressed in the gas token (`$APT` for Aptos-Move). A transaction specifies the value $GasPrice(tx)$ it is willing to pay.  
+The value $GasPrice(tx)$ must not be lower than a minimum $GasPrice$, which may fluctuate depending on network contention, and may be updated frequently (e.g. after each block). 
+
+In contrast, the permanent storage fees are designed to be stable and are updated infrequently. The reasoning is that they depend on how much storage costs (hardware, disks) and with advancement in technology this should go down in the future. The storage fees for $tx$ are denoted $StoreFees(tx)$ and expressed in `$APT`[^3].
+
+The fees that correspond to the permanent storage are **converted to gas units** using the  $GasPrice$ and the `$APT` price. 
+
+The _total charge in gas units_ for a transaction is the defined by:
+
+$$ TotalGasUnits(tx) = g(tx) + StoreFees(tx)/GasPrice(tx)$$
+
+and the total transaction fees are approximately[^1]:
+
+$$ ExecFees(tx) = TotalGasUnits(tx) \times GasPrice(tx) \mathpunct.$$
+
+
+> [!WARNING]
+We may not use the `$APT` token value to compute the fees on Movement. If we do so the $ExecFees$ will be identical to processing fees on Aptos-Move, and the total fees including $DAFees$ and $SetlFees$ will be higher than the processing fees on Aptos-Move, which may not be desirable.
+
+
+> [!TIP]
+**Proposal 1**: use the $TotalGasUnits(tx)$ and a $GasPrice$ in `$MOVE` to compute the execution fees on a Movement chain.
+This may require an oracle to get the exchange rate for `$APT`/`$MOVE`.
```

Comment:
i dont understand the reasoning for this. How does $APT come into this? - the gasprice is in $MOVE